recipes/m4/all/conanfile.py

```diff
@@ -26,8 +26,10 @@ def _is_clang(self):
         return str(self.settings.compiler).endswith("clang")
 
     def build_requirements(self):
-        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ and \
-                tools.os_info.detect_windows_subsystem() != "msys2":
+        if (tools.os_info.is_windows
+            and "CONAN_BASH_PATH" not in os.environ
+            and ((tools.os_info.detect_windows_subsystem() != "msys2")
+                 or not tools.which("make"))):
```

Comment:
I don't feel really happy with this change.
@ericLemanissier How about we just do the following?
```suggestion
        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```

If a user on msys2 really does not want to use cci's msys2, then he should set `CONAN_BASH_PATH` in his profile.

Reply:
it should be good enough

Reply:
Would you be ok with doing that for all recipes?

Reply:
Btw, I just tried a building a recipe on a installed msys2 (non-conan package).
`self.run(..., winbash=tools.get_env.is_windows)` commands fail with
`ConanException: Command only for Windows operating system`.

Reply:
I would be ok with this change for all recipes

Reply:
Should I go ahead and commit @madebr 's suggestion above then?